misc/rules/types.yml

```diff
@@ -0,0 +1,40 @@
+tosca_definitions_version: tosca_simple_yaml_1_3
+
+node_types:
+  sodalite.nodes.Configuration.ruleset:
+    derived_from: tosca.nodes.SoftwareComponent
+    description: Node for registration/deregistration of the ruleset file
+    properties:
+      ruleset:
+        type: list
```

Comment:
I think a rule (element of ruleset list) should be defined as datatype, so it can be validated, which would ensure the existence of proper input params (name, expr, for etc.)

Take a look at the implementation of Openstack security rules module ([node definitions](https://github.com/SODALITE-EU/iac-modules/blob/20da61819e07c576cf610fdfd296fce4cf11c64b/openstack/openstack_security_rule.yaml), [usage](https://github.com/SODALITE-EU/xopera-rest-api/blob/048397af1a1da9b1924b5abdc3c6b75fc9e24db2/xOpera-rest-blueprint/service.yaml#L70))

Reply:
I have implemented the datatypes

Reply:
Datatype looks good hovewer ruleset.yml fails if desc or summary params are missing.

Every parameter should either (1) be required or (2) have default value defined (so it is used if params is omitted and templating does not fail)